api/v1beta1/agentserviceconfig_types.go

```diff
@@ -102,6 +102,12 @@ type AgentServiceConfigSpec struct {
 	// +kubebuilder:validation:default:=disabled
 	// +optional
 	IPXEHTTPRoute string `json:"iPXEHTTPRoute,omitempty"`
+
+	// KubeconfigSecretRef is the reference to the Secret containing a kubeconfig of an external
+	// control plane (e.g. hypershift HostCluster).
+	// +optional
+	//+operator-sdk:csv:customresourcedefinitions:type=spec,displayName="Control plane kubeconfig reference"
+	KubeconfigSecretRef *corev1.LocalObjectReference `json:"kubeconfigSecretRef"`
```

Comment:
Per discussion earlier today, we need to further discuss what the right API is for this and to signal the other behavioral changes that are needed for the hypershift case.